manifests/haproxy/autoload.pp

```diff
@@ -28,47 +32,106 @@
         replace => false,
       }
 
-      $http_edge_redirect_rules = flatten($http_edge_domains_envs.map |$domain, $options| {
-        $match_type = pick($options['match_type'], 'str')
-        $hdr_match = $match_type ? {
-          'str' => 'hdr',
-          default => "hdr_${match_type}"
-        }
-        $domain_separator = $match_type ? {
-          'reg' => '\.',
-          default => '.',
-        }
-        $options['_envs'].map |$env| {
-          $path_env = "${http_edge_path}/${env}"
-          if $options[$env] {
-            $domain_env = $options[$env]
+      $http_edge_rules_all_placeholder = $http_edge_services.map | $service | {
+        $http_edge_rules = flatten($http_edge_domains_envs.map |$domain, $options| {
+          $match_type = pick($options['match_type'], 'str')
+          $hdr_match = $match_type ? {
+            'str' => 'hdr',
+            default => "hdr_${match_type}"
           }
-          else {
-            $parts = [$options['_domain_prefix'], $env, $options['_domain_suffix']].filter |$item| { !$item.empty }
-            $domain_env = join($parts, $domain_separator)
+          $domain_separator = $match_type ? {
+            'reg' => '\.',
+            default => '.',
           }
+          $options['_envs'].map |$env| {
+            $path_env = "${http_edge_path}/${env}"
+            if $options[$env] {
+              $domain_env = $options[$env]
+            }
+            else {
+              $parts = [$options['_domain_prefix'], $env, $options['_domain_suffix']].filter |$item| { !$item.empty }
+              $domain_env = join($parts, $domain_separator)
+            }
+
+            if $service in $options['_services'] {
```

Comment:
@jonhattan I suspect this is the cause of the empty http-requests, in cases where the condition is not met, the map does not return a particular string, which I am guessing it is interpreted as an empty string. 

I am trying to use filter() to remove the empty strings, however if there is a way to tell the map() function to return a null value it would be the optimal solution, since we would only hace to add an else conditional

Reply:
resolver